src/api/user/src/routes/user.js

```diff
@@ -0,0 +1,139 @@
+const { Router, logger } = require('@senecacdot/satellite');
+const { errors } = require('celebrate');
+const celebrateSchema = require('../models/celebrateSchema');
+const User = require('../models/user');
+const db = require('../services/firestore');
+
+const router = Router();
+
+// get a user with a supplied id, validated by the celebrateSchema
+// rejected if a user could not be found, otherwise user returned
+router.get('/:id', celebrateSchema.get(), async (req, res, next) => {
+  try {
+    const userRef = db.collection('users').doc(req.params.id);
+    const doc = await userRef.get();
+
+    if (!doc.exists) {
+      logger.debug(`User data (id: ${doc.id}) was requested by ${req.ip} but could not be found.`);
+      res.status(404).json({
+        msg: `User data (id: ${doc.id}) was requested but could not be found.`,
+      });
+    } else {
+      logger.debug(`User data (id: ${doc.id}) was requested by ${req.ip} and served successfully.`);
+      res.status(200).json(doc.data());
+    }
+  } catch (err) {
+    next(err);
+  }
+});
+
+// get all users
+// rejected if the user collection is empty, otherwise users returned
+router.get('/', async (req, res, next) => {
+  try {
+    const usersRef = db.collection('users');
+    const snapshot = await usersRef.get();
+
+    if (snapshot.empty) {
+      logger.debug(`All users were requested by ${req.ip} but could not be found.`);
+      res.status(404).json({
+        msg: `That collection could not be found, or contains no data.`,
+      });
+    } else {
+      logger.debug(`All users were requested by ${req.ip} and served successfully.`);
+
+      const usersArray = [];
+      snapshot.forEach((doc) => {
+        usersArray.push(doc.data());
+      });
+
+      res.status(200).json(usersArray);
+    }
+  } catch (err) {
+    next(err);
+  }
+});
+
+// post a user with supplied info, validated by the celebrateSchema
+// rejected if a user already exists with that id, otherwise user created
+router.post('/', celebrateSchema.post(), async (req, res, next) => {
```

Comment:
I filed https://github.com/Seneca-CDOT/telescope/issues/1931 and some related things to handle securing these routes.